src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health) {
+    this.health = health || 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+  static updateAlive() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(harbivore) {
+    if (!harbivore.hidden && harbivore instanceof Herbivore) {
+      harbivore.health -= 50;
+    }
+    Animal.updateAlive();
```

Comment:
The method 'Animal.updateAlive' is correctly used to filter out dead animals. However, it would be more efficient to call this method only if the health of the herbivore actually changes, i.e., inside the if statement after reducing the health.